### ISSUE TITLE
srars for survreg object

### ISSUE THREAD

**Issue by pviton** [leifeld/texreg#26]
[message also sent directly to p. leifeld]

`I think there may be something wrong with the
way texreg calculates stars on these objects. Here's
the result of a survreg estimation run:

```
Call:
survreg(formula = Surv(N_Days, censDecSummer) ~ 1 + chief + CrimX + 
    CivLibX + EconX + FirstAmX + Total_Amicus + DuePX + AttysX + 
    UnionsX + JudPowX + FedismX + TaxX + cluster(term) + percuriam + 
    nDissents + nConcurs + nCresults + nytSalience + declarationUncon, 
    data = timing, subset = nz, dist = "weibull")
                   Value Std. Err (Naive SE)       z         p
(Intercept)       4.2156  0.11722    0.07189  35.962 3.23e-283
chiefRehnquist   -0.2313  0.04735    0.02478  -4.883  1.04e-06
chiefRoberts     -0.2146  0.05736    0.03802  -3.741  1.83e-04
chiefVinson      -0.1704  0.09280    0.03114  -1.836  6.63e-02
chiefWarren      -0.3470  0.06112    0.02423  -5.677  1.37e-08
CrimXTRUE        -0.0705  0.09471    0.06245  -0.745  4.57e-01
CivLibXTRUE      -0.0609  0.08270    0.06318  -0.737  4.61e-01
EconXTRUE        -0.0878  0.08948    0.06223  -0.981  3.27e-01
FirstAmXTRUE      0.0577  0.10986    0.06924   0.525  6.00e-01
Total_Amicus      0.0195  0.00408    0.00257   4.775  1.79e-06
DuePXTRUE         0.0823  0.11426    0.07440   0.720  4.71e-01
AttysXTRUE        0.0455  0.11492    0.10697   0.396  6.92e-01
UnionsXTRUE      -0.1243  0.11907    0.07111  -1.044  2.97e-01
JudPowXTRUE      -0.0410  0.09329    0.06360  -0.440  6.60e-01
FedismXTRUE      -0.1513  0.08744    0.07121  -1.731  8.35e-02
TaxXTRUE         -0.0961  0.12706    0.07250  -0.756  4.50e-01
percuriam        -0.3020  0.07135    0.03308  -4.232  2.31e-05
nDissents         0.1637  0.00905    0.00611  18.097  3.34e-73
nConcurs          0.1084  0.01498    0.01187   7.237  4.60e-13
nCresults         0.1583  0.01771    0.01083   8.937  3.98e-19
nytSalience       0.2927  0.05540    0.02876   5.284  1.26e-07
declarationUncon  0.0743  0.02610    0.01972   2.846  4.43e-03
Log(scale)       -0.4851  0.03711    0.00985 -13.074  4.66e-39

Scale= 0.616 

Weibull distribution
Loglik(model)= -28011.6   Loglik(intercept only)= -29009.1
	Chisq= 1995.02 on 21 degrees of freedom, p= 0 
(Loglikelihood assumes independent observations)
Number of Newton-Raphson Iterations: 5 
n=7387 (9 observations deleted due to missingness)
```

and here's what happens when I run this through screenreg:
(etr1 is the result of extract.survreg)

```
> screenreg(etr1,single.row=T,digits=3,custom.model.names="Model 1")

========================================
                  Model 1               
----------------------------------------
(Intercept)            4.216 (0.117)    
chiefRehnquist        -0.231 (0.047) ***
chiefRoberts          -0.215 (0.057) ***
chiefVinson           -0.170 (0.093) ***
chiefWarren           -0.347 (0.061) ***
CrimXTRUE             -0.071 (0.095) ***
CivLibXTRUE           -0.061 (0.083) ***
EconXTRUE             -0.088 (0.089) ***
FirstAmXTRUE           0.058 (0.110)    
Total_Amicus           0.019 (0.004)    
DuePXTRUE              0.082 (0.114)    
AttysXTRUE             0.045 (0.115)    
UnionsXTRUE           -0.124 (0.119) ***
JudPowXTRUE           -0.041 (0.093) ***
FedismXTRUE           -0.151 (0.087) ***
TaxXTRUE              -0.096 (0.127) ***
percuriam             -0.302 (0.071) ***
nDissents              0.164 (0.009)    
nConcurs               0.108 (0.015)    
nCresults              0.158 (0.018)    
nytSalience            0.293 (0.055)    
declarationUncon       0.074 (0.026)    
Log(scale)            -0.485 (0.037) ***
----------------------------------------
AIC                56069.214            
Log Likelihood    -28011.607            
========================================
*** p < 0.001, ** p < 0.01, * p < 0.05
```

The stars look wrong to me, at least if they're being
calcuated as coeff/std-error, which appears to be represented
as pvalue in the extract.survreg result. To check this, I did:

```
> c1<-etr1@coef	   # extract coeffs
> se1<-etr1@se     # extract se s
> t1<-c1/se1	   # calculate tstats
> p1<-etr@pvalues  # extract pvalues, for comparison
> cbind(c1,se1,t1,p1)
                          c1         se1          t1          p1
(Intercept)       4.21563707 0.117223289  35.9624535  35.9624535
chiefRehnquist   -0.23125223 0.047354249  -4.8834526  -4.8834526
chiefRoberts     -0.21459294 0.057360414  -3.7411331  -3.7411331
chiefVinson      -0.17039978 0.092795972  -1.8362842  -1.8362842
chiefWarren      -0.34695761 0.061116736  -5.6769655  -5.6769655
CrimXTRUE        -0.07052040 0.094711720  -0.7445795  -0.7445795
CivLibXTRUE      -0.06094230 0.082695655  -0.7369468  -0.7369468
EconXTRUE        -0.08775338 0.089477503  -0.9807313  -0.9807313
FirstAmXTRUE      0.05768771 0.109862080   0.5250921   0.5250921
Total_Amicus      0.01948112 0.004079518   4.7753480   4.7753480
DuePXTRUE         0.08227394 0.114257493   0.7200748   0.7200748
AttysXTRUE        0.04548361 0.114924076   0.3957710   0.3957710
UnionsXTRUE      -0.12430357 0.119071832  -1.0439377  -1.0439377
JudPowXTRUE      -0.04100121 0.093290039  -0.4395026  -0.4395026
FedismXTRUE      -0.15133097 0.087437920  -1.7307247  -1.7307247
TaxXTRUE         -0.09606226 0.127062616  -0.7560230  -0.7560230
percuriam        -0.30197049 0.07136025  -4.2324781  -4.2324781
nDissents         0.16370191 0.009045561  18.0974865  18.0974865
nConcurs          0.10842433 0.014982691   7.2366394   7.2366394
nCresults         0.15831111 0.017713210   8.9374602   8.9374602
nytSalience       0.29272654 0.055398539   5.2840120   5.2840120
declarationUncon  0.07429407 0.026103381   2.8461475   2.8461475
Log(scale)       -0.48509865 0.037105383 -13.0735386 -13.0735386
```

Note that the last two columns are the same. What I don't see
is why chiefRehnquist, with a pvalue of 4.88 gets 3 stars, while
nConcurs, with a pvalue of 8.94 gets none. Or why nDissents,
with a pvalue of 18.10 also gets none. Am I missing something
here? (Now that I look at it, it seems that only negative
pvalues are being correctly starred: could there be a problem
with positive values?).
`

**Comment by leifeld** [comment]
Thanks for reporting this. Can you provide a minimal self-contained example? I tried to reproduce the problem with the examples in the `survreg` help file, but they seem to be fine. At least after some other minor fixes I just committed.

**Comment by pviton** [comment]
Sure. I've attached the dataset as timing.zip (contains TimingData.rdata).

In R, load in the dataset (called 'timing')
and define the following function, which sets up some defaults
a bit differently. (See email version of my post).	  Then

========================== R session ======================

extract_tr<-function(mod){extract.survreg(mod,include.nobs=F,include.bic=F,include.deviance=F)}

library(survival)
library(texreg)

nz<-timing$N_Days > 0 # censoring indicator

mod1<-survreg(Surv(N_Days,censDecSummer)~1+chief+CrimX+CivLibX+EconX+FirstAmX+Total_Amicus+
								DuePX+AttysX+UnionsX+JudPowX+FedismX+TaxX+cluster(term)+percuriam+
								nDissents+nConcurs+nCresults+nytSalience+declarationUncon,
							data=timing,subset=nz,dist="weibull")
							
summary(mod1) # should reproduce the model in the email/posting


etr1<-extract_tr(mod1)
screenreg(mod1)


=========================================================

Incidentally, I've made a bit of progress on the defaults issue
(mentioned only in the email as it's not critical); I'll
email separately on this when I'm sure.

[timing.zip](https://github.com/leifeld/texreg/files/748664/timing.zip)

**Comment by pviton** [comment]
Additional comment: when the user specifies 'cluster' in the formula (as I do in my example), the table created by the survreg summary has 5 columns, not 4. I think you want to ensure that pval as used in texreg is the last column, however many there are. Would this solve the problem?

**Comment by leifeld** [comment]
I didn't get any of the e-mails you sent me. Did you use my e-mail address at the University of Glasgow? Please resend or post here on the issue tracker.

I fixed the issue in the package. Check out this version here: [texreg_1.36.22.tar.gz](https://github.com/leifeld/texreg/files/749315/texreg_1.36.22.tar.gz)

The main problem was indeed that in this case there were five table columns and the function was using the fourth. I changed that bit. I also had to fix the use of the `nobs` function in the code because that is apparently no longer supported by the `survival` package.

**Comment by pviton** [comment]
In my email (which I did send to glasgow, so I'm not sure why
it misfired) I mentioned that there were problems with the nobs,
bic and deviance defaults: in the old version, unless these
were set to FALSE, texreg wouldn't work with a survreg object.

nobs   You've fixed this in 1.36.22; though I'd consider using
length(model$linear.predictors) which is the way it's
done in summary.survreg, just to be on the safe side.

bic : Also fixed by calling BIC(model).


deviance : deviance(model) returns NULL, so this is taken care of
in the extract.survreg code.

So it looks as though all problems with survreg may be fixed.
Will you be posting a compiled version of the package soon?

**Comment by leifeld** [comment]
Thanks, I have changed the way the number of observations is retrieved in c20757dac2219687aeaa0c458266613c784befd9. I'll push a new release to CRAN in a couple of weeks. I need to fix some other issues first... But you can build it from source any time.
